File: Week4/面试题 04.06. 后继者.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderSuccessor(self, root: TreeNode, p: TreeNode) -> TreeNode:
        res=[]
        def order(root):
            if root:
                order(root.left)
                res.append(root)
                order(root.right)
        order(root)
        index=res.index(p)
        if index==len(res)-1:
            return
        else:
            return res[index+1]

File: Week4/面试题 04.06. 后继者.js
var inorderSuccessor = function(root, p) {
    var res=null
    var Successor=function(root,node){
        if (root!=null && node!=null){
            if(root.val>node.val){
                if (res==null){
                    res=root
                }else if(res.val>root.val){
                    res=root
                }
                Successor(root.left,node)
            }else if (root.val==node.val){
                if (root.right!=null){
                    cur=root.right
                    while(cur.left!=null){
                        cur=cur.left
                    }
                    res=cur
                }
            }else{
                Successor(root.right,node)
            }
        }
    }
    Successor(root,p)
    return res
};